internal/prometheusbp/spectest/spec.go

```diff
@@ -168,16 +191,37 @@ func buildLables(name, prefix, clientOrServer string) map[string]struct{} {
 	default:
 		labelSuffixes = []string{}
 	}
+	return labelSuffixes
+}
 
-	if clientOrServer == client {
-		labelSuffixes = append(labelSuffixes, "slug")
-	}
-
-	var wantLabels = map[string]struct{}{}
-	for _, label := range labelSuffixes {
-		wantLabels[prefix+"_"+label] = struct{}{}
+// grpcSpecificLabels returns the following labels:
+// latency_seconds metrics expect the following labels:
+//   - "service"
+//   - "method"
+//   - "success"
+//   - "type"
+// requests_total metrics expect the following labels:
+//   - "service"
+//   - "method"
+//   - "success"
+//   - "type"
+//   - "code"
+// active_requests metrics expect the following labels:
+//   - "service"
+//   - "method"
+func grpcSpecificLabels(name string) []string {
+	labelSuffixes := []string{"service", "method"}
+	switch {
+	case strings.HasSuffix(name, "_latency_seconds"):
+		labelSuffixes = append(labelSuffixes, "type", "success")
+	case strings.HasSuffix(name, "_requests_total"):
+		labelSuffixes = append(labelSuffixes, "type", "success", "code")
+	case strings.HasSuffix(name, "_active_requests"):
+		// no op
+	default:
+		labelSuffixes = []string{}
```

Comment:
I would probably say "return nil" if you want to emphasize that you are "ignoring" labels for things that don't match